text/chapter4.md

```diff
@@ -314,9 +314,9 @@ The keyword `do` introduces a block of code which uses do notation. The block co
 - Expressions which do not bind elements of the array to names. The last line `pure [i, j]` is an example of this kind of expression.
 - Expressions which give names to expressions, using the `let` keyword.
 
-This new notation hopefully makes the structure of the algorithm clearer. If you mentally replace the arrow `<-` with the word "choose", you might read it as follows: "choose an element `i` between 1 and n, then choose an element `j` between `i` and `n`, and return `[i, j]`".
+This new notation hopefully makes the structure of the algorithm clearer. If you mentally replace the arrow `<-` with the word "choose", you might read it as follows: "choose an element `i` between 1 and n, then choose an element `j` between `i` and `n`, and return the desired results, `i` and `j`, as `[i, j]`.
```

Comment:
I don't understand what this is adding. If you know what "return" means in the context of programming, which I'm pretty sure is something we are assuming about the reader, the fact that the things we are returning are the desired results is kind of already obvious? What confusion would this prevent?

Reply:
I confess that I tripped myself up by confusing the result array with the array in the problem itself.  If you look at the notes above, @milesfrain and I discussed perhaps using a different type such as `Tuple` or `Data.Pair` for the return value to prevent readers like me from essentially doubting what `return` meant in this context.  I'm hoping that my text change emphasizes that what we're interested in is the results for `I` and `j`, not how we package them to return.  Ideally, returning `I` and `j` in a `Data.Pair` would be the cleanest way to return the result, but, as I said above, learning about `Data.Pair` is too much to additionally learn at this point in the book.